scipy/sparse/csc.py

```diff
@@ -212,12 +203,29 @@ def getcol(self, i):
             i += N
         if i < 0 or i >= N:
             raise IndexError('index (%d) out of range' % i)
-        idx = slice(*self.indptr[i:i+2])
-        data = self.data[idx].copy()
-        indices = self.indices[idx].copy()
-        indptr = np.array([0, len(indices)], dtype=self.indptr.dtype)
-        return csc_matrix((data, indices, indptr), shape=(M, 1),
-                          dtype=self.dtype, copy=False)
+        return self._get_submatrix(major=i, copy=True)
+
+    def _get_intXarray(self, row, col):
+        return self._major_index_fancy(col)._get_submatrix(minor=row)
+
+    def _get_intXslice(self, row, col):
+        if col.step in (1, None):
+            return self._get_submatrix(major=col, minor=row, copy=True)
+        return self._major_slice(col)._get_submatrix(minor=row)
```

Comment:
Would it be more efficient to go through arrayXarray? I'm a bit worried seeing that many of these create temporary matrices first slicing along one axis and then along the other, which probably involves data copies right?

Reply:
This particular case is a bit tricky:
 * We avoid a full copy when creating the intermediate matrix, because `_major_slice` only needs to allocate a new `indptr` array. The other two arrays are left as views into the original matrix.
 * We end up re-allocating the index and data views during the `_get_submatrix` call, because it routes through a sparsetools routine that ends up converting to a contiguous C array.

In general, I tried to be careful to avoid needless allocations in all the `_`-prefixed internal functions, but as in this case we do sometimes pay for temporaries. The good news is that, after this PR, it should be relatively simple to define more efficient indexing methods for the specific cases that turn out to be most important to optimize further.